lib/Frontend/Frontend.cpp

```diff
@@ -559,6 +559,23 @@ void CompilerInstance::setupCachingDiagnosticsProcessorIfNeeded() {
   CDP->startDiagnosticCapture();
 }
 
+void CompilerInstance::setUpVFSForDirectClangCC1EmitPCM() {
+  // Set up the VFS overlay so that the -direct-clang-cc1-module-build
+  // emit-pcm command can find a system modulemap file (like
+  // vcruntime.modulemap) as an input file.
+  if (Invocation.getClangImporterOptions().DirectClangCC1ModuleBuild &&
+      Invocation.getFrontendOptions().RequestedAction ==
+          FrontendOptions::ActionType::EmitPCM) {
+    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS =
+        SourceMgr.getFileSystem();
+    ClangInvocationFileMapping FileMapping = applyClangInvocationMapping(
+        *Context, nullptr, VFS, /*suppressDiagnostic=*/false);
+    if (!FileMapping.redirectedFiles.empty()) {
```

Comment:
I don't think this is correct for caching. Caching build always uses direct clang cc1 mode but the input is include-tree, which doesn't have any notion of VFS anymore.